src/ppom-controller.ts

```diff
@@ -332,25 +342,42 @@ export class PPOMController extends BaseControllerV2<
    * This function receives a callback that will be called with the PPOM.
    *
    * @param callback - Callback to be invoked with PPOM.
+   * @param networkClientId - Optional network client ID to use for the PPOM requests.
    */
   async usePPOM<T>(
-    callback: (ppom: any) => Promise<T>,
+    callback: (ppom: unknown) => Promise<T>,
+    networkClientId?: NetworkClientId,
   ): Promise<T & { providerRequestsCount: Record<string, number> }> {
     if (!this.#securityAlertsEnabled) {
       throw Error('User has securityAlertsEnabled set to false');
     }
-    if (!this.#networkIsSupported(this.#chainId)) {
-      throw Error('Blockaid validation is available only on ethereum mainnet');
+
+    let chainId = this.#chainId;
+    let provider = this.#provider;
+    if (networkClientId) {
+      const networkClient = this.messagingSystem.call(
+        'NetworkController:getNetworkClientById',
+        networkClientId,
+      );
+      chainId = networkClient.configuration.chainId;
+      provider = networkClient.provider;
     }
+    this.#updateChainStatus(chainId);
 
-    await this.#getNewFilesForCurrentChainIfNeeded();
+    if (!this.#networkIsSupported(chainId)) {
+      throw Error('Blockaid validation is available only on ethereum mainnet');
+    }
```

Comment:
[nit] does it make sense to throw this error before updating the chain status?

Reply:
i went back and forth on this haha. This depends on what it means for a network/chain to be "seen". I understand that it doesn't make sense to update the state for the chainId if it's not even supported, but that is actually what currently happens already with the single globally selected network

Reply:
@jpuri what do you think?

Reply:
+1 to @jiexi

Reply:
https://github.com/MetaMask/ppom-validator/pull/85/commits/d8420162f2c73be3acb89aa6e1e9cece0539ed2a